subnet/add_validator_subnet.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package subnet
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/avalanche"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/multisig"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/wallet"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"golang.org/x/net/context"
+)
+
+type ValidatorParams struct {
+	NodeID ids.NodeID
+
+	Duration time.Duration
+
+	Weight uint64
+
+	Network avalanche.Network
+}
+
+// AddValidator adds validator to subnet
+// Before an Avalanche Node can be added as a validator to a Subnet, the node must already be
+// tracking the subnet
+// TODO: add more description once node join subnet sdk is done
+func (c *Subnet) AddValidator(wallet wallet.Wallet, validatorInput ValidatorParams) (*multisig.Multisig, error) {
+	if validatorInput.NodeID == ids.EmptyNodeID {
+		return nil, fmt.Errorf("validator node id is not provided")
+	}
+	if validatorInput.Duration == 0 {
+		return nil, fmt.Errorf("validator duration is not provided")
+	}
+	if validatorInput.Weight == 0 {
+		return nil, fmt.Errorf("validator weight is not provided")
+	}
+	if c.SubnetID == ids.Empty {
+		return nil, fmt.Errorf("subnet ID is not provided")
+	}
+
+	wallet.SetSubnetAuthMultisig(c.DeployInfo.SubnetAuthKeys)
```

Comment:
do we want this to depend on network? probably

Reply:
I mean, c.DeployInfo.SubnetAuthKeys is network dependent, may it be the case that the user mix wallet networks and subnet auth keys?